src/librustc_codegen_ironox/metadata.rs

```diff
@@ -1,6 +1,7 @@
 extern crate ar;
+extern crate goblin;
```

Comment:
Interesting. I used the `elf` crate for ELF parsing. I'd be interested to know which is better. In the long run we don't want to pull in two ELF parsers (when my work is merged with yours one day).

Reply:
I'll keep that in mind. I'll rewrite the metadata loader to use `elf` if it turns out to be better than `goblin`. I only used `goblin` because it seemed to be more popular.

Reply:
I wouldn't worry about it just now. It shouldn't be hard to convert at a later date.

It looks like `goblin` is more portable and supports PE and MACH binaries, as well as ELF, so perhaps we do want that if we are to run on Windows/OSX.